### PR TITLE
CDB list input validation

### LINKED ISSUE #1252: CDB List should validate white spaces in keys
| Wazuh | Splunk | Rev |
| ----- | ------- | --- |
| 4.3   | 8.2.4 | 444 |

**Description**
Create new CDB List form should validate white spaces in keys to prevent server side errors.

**Steps to reproduce**

1. Go to `Management / Ruleset`
2. Click on `Lists` tab
3. Click on `Add new list`
4. Fill in a file name without spaces
5. Create keys and values with spaces
6. Click Save file button and see error
. 
**Screenshots**
image

image

**Additional context**
Add any other context about the problem here. Here you can paste log entries too or any other useful information that may help with the issue.

### PR DESCRIPTION
**Pull request by AlexRuiz7**
Summary
---------
Closes #1252 

This PR adds input data validation for the CDB lists.

- Validates the CDB list name by using this regular expression: `^[\\w\\.\\-]+$`
- Validates the CDB key:value pairs
  - Keys: `?:^"([\\w\\-:]+?)"|^[^:"\\s]+$)`
  - Values: `(?:^"([\\w\\-:]*?)"$|^[^:"]*$)`

This PR also renames the CDB List navigation item from 'Lists' to 'CDB Lists' to make it more descriptive and consistent with the Kibana App.

To test
-------

- Check that the navigation item to go to the CDB lists is named as 'CDB Lists' across all subsections of the Ruleset section.
- Check that CDB list with invalid name cannot be created.
    - Can only contain alphanumeric characters, dots (.), hyphens (-) and underscores (_).
- Check that no key:pair value can be created using an invalid **key** (invalid values are replaced with an empty string).
  - Examples [here](https://github.com/wazuh/wazuh/blob/eaab105fb5a7430173a52a2598deef204b06e809/framework/wazuh/core/cdb_list.py#L185).

### CODE REVIEW

**Review by asteriscos — CHANGES_REQUESTED** [review]
Beware CBD list throws an error when it has a `.` character and the input validation allows it.

image

**Review by asteriscos — APPROVED** [review]
CR: :heavy_check_mark: 
Test: :heavy_check_mark:

File: SplunkAppForWazuh/appserver/static/js/controllers/management/cdb/managerCdbCtrl.js [hunk 1]
```diff
@@ -59,6 +59,11 @@ define([
      * On controller load
      */
     $onInit() {
+      // Data validation
+      this.scope.nameValidationRegex = '^[\\w\\-]+$'
+      this.scope.keyValidationRegex = '(?:^"([\\w\\-:]+?)"|^[^:"\\s]+$)'
+      this.scope.valueValidationRegex = '(?:^"([\\w\\-:]*?)"$|^[^:"\\s]*$)'
```
Comment (Desvelao) [inline SplunkAppForWazuh/appserver/static/js/controllers/management/cdb/managerCdbCtrl.js:65]: **question**: where do these regular expressions come from?

Comment (AlexRuiz7) [reply SplunkAppForWazuh/appserver/static/js/controllers/management/cdb/managerCdbCtrl.js:65]: From the Wazuh API --> https://github.com/wazuh/wazuh/blob/eaab105fb5a7430173a52a2598deef204b06e809/framework/wazuh/core/cdb_list.py#L185

**Review by Desvelao — COMMENTED** [review]
**review**

Test :x:

I found some things that I am not sure if there is a problem:
- The value seems can't contain spaces, but there is built-in list with values with spaces, for example `aws-eventnames`
image
image

File: SplunkAppForWazuh/appserver/static/js/controllers/management/cdb/managerCdbCtrl.js (hunk 1 above)
Comment (Desvelao) [reply SplunkAppForWazuh/appserver/static/js/controllers/management/cdb/managerCdbCtrl.js:65]: **nitpick**: Check the regex, the `valueValidationRegex` is not the equal. See https://github.com/wazuh/wazuh/blob/v4.3.0-rc4/framework/wazuh/core/cdb_list.py#L203. The regex in the app invalidates the `values` with spaces at least.

Comment (AlexRuiz7) [reply SplunkAppForWazuh/appserver/static/js/controllers/management/cdb/managerCdbCtrl.js]: Solved

**Comment by Machi3mfl** [comment]
Possible improvement:
Screen Shot 2022-02-16 at 13 44 56

Change the "Add new entry" style when is disabled.

**Review by Machi3mfl — APPROVED** [review]
CR ✅
Test ✅

**Pull request merged**